src/orquestra/quantum/circuits/_gates.py

```diff
@@ -349,7 +349,7 @@ def num_qubits(self) -> int:
 
     @property
     def name(self):
-        return DAGGER_GATE_NAME
+        return self.wrapped_gate.name + "_" + DAGGER_GATE_NAME
```

Comment:
[question] Actually, shouldn't we refactor all of the operations like this?

Reply:
I thought of doing this, but it would be a bit harder. For instance, in the controlled gate case, are we just going stack controls on top of one another? This would create a very large gate name if you have a lot of controls. For the exponent, should we include the exponent in the name? There's a lot of choices to be made here and I'm not sure the best way to do it.

Reply:
We would also have to make sure that this works with all our other conversions too, so I would say adding this functionality to other wrappers is out of scope for now.

Reply:
Sounds fine 👌